### PR TITLE
Fixes Implement Zod Validation for /logout_all API Route #138

### PR DESCRIPTION
**Pull request by Pravesh-Sudha**
Issue #138

### CODE REVIEW

File: packages/server/src/api_auth/index.ts [hunk 1]
```diff
@@ -6,6 +6,11 @@ import { emailPasswordValidator, emailValidator } from '../validators';
 const rateLimit = require('express-rate-limit');
 
 const router = Router();
+const logoutAllSchema = z.object({
+    token: z.string().min(1), // Ensure the token property is a non-empty string
+    email: z.string().min(1), // Ensure the email property is a non-empty string
```
Comment (prasenjeet-symon) [inline packages/server/src/api_auth/index.ts:11]: We can more specific here ( email validator )

File: packages/server/src/api_auth/index.ts [hunk 2]
```diff
@@ -6,6 +6,11 @@ import { emailPasswordValidator, emailValidator } from '../validators';
 const rateLimit = require('express-rate-limit');
 
 const router = Router();
+const logoutAllSchema = z.object({
```
Comment (prasenjeet-symon) [inline packages/server/src/api_auth/index.ts:9]: We can put this validator in the validators folder.

File: packages/server/src/api_auth/index.ts [hunk 3]
```diff
@@ -507,6 +512,15 @@ router.post('/logout', authenticateUser, async (req, res) => {
...
+        // Validate the request body using the Zod schema
+        logoutAllSchema.parse(req.locals);
+    } catch (error) {
+        if (error instanceof ZodError && !error.isEmpty) {
+            res.status(400).send({ error: "Token and email are required and must be non-empty" });
+            return;
+        }
+    }
     // res.locals should contain email and token
     if (!('token' in res.locals) || !('email' in res.locals)) {
```
Comment (prasenjeet-symon) [inline packages/server/src/api_auth/index.ts:525]: We can delete this block and all other old validation logic.

**Review by prasenjeet-symon — CHANGES_REQUESTED** [review]
Please consider the changes and re-submit the PR. Thanks so much

File: packages/server/src/api_auth/index.ts (hunk 2 above)
Comment (Pravesh-Sudha) [reply packages/server/src/api_auth/index.ts:9]: I will put it in validator's index.js file

File: packages/server/src/api_auth/index.ts (hunk 3 above)
Comment (Pravesh-Sudha) [reply packages/server/src/api_auth/index.ts:525]: Ok, will do it shortly

File: packages/server/src/api_auth/index.ts (hunk 1 above)
Comment (Pravesh-Sudha) [reply packages/server/src/api_auth/index.ts:11]: for email that it should include @ and .com/.... types properties?

**Pull request closed without merging**